roles/kubernetes/preinstall/tasks/0040-verify-settings.yml

```diff
@@ -63,40 +53,23 @@
     msg: "The current release of Kubespray only support newer version of Kubernetes than {{ kube_version_min_required }} - You are trying to apply {{ kube_version }}"
   when: not ignore_assert_errors
 
-# simplify this items-list when   https://github.com/ansible/ansible/issues/15753  is resolved
 - name: "Stop if known booleans are set as strings (Use JSON format on CLI: -e \"{'key': true }\")"
   assert:
-    that: item.value | type_debug == 'bool'
-    msg: "{{ item.value }} isn't a bool"
+    that:
+      - download_run_once | type_debug == 'bool'
+      - deploy_netchecker | type_debug == 'bool'
+      - download_always_pull | type_debug == 'bool'
+      - helm_enabled | type_debug == 'bool'
+      - openstack_lbaas_enabled | type_debug == 'bool'
   run_once: true
-  with_items:
-    - { name: download_run_once, value: "{{ download_run_once }}" }
-    - { name: deploy_netchecker, value: "{{ deploy_netchecker }}" }
-    - { name: download_always_pull, value: "{{ download_always_pull }}" }
-    - { name: helm_enabled, value: "{{ helm_enabled }}" }
-    - { name: openstack_lbaas_enabled, value: "{{ openstack_lbaas_enabled }}" }
   when: not ignore_assert_errors
 
 - name: Stop if even number of etcd hosts
   assert:
-    that: groups.etcd | length is not divisibleby 2
-  when:
-    - not ignore_assert_errors
-    - inventory_hostname in groups.get('etcd',[])
-
-- name: Stop if memory is too small for control plane nodes
-  assert:
-    that: ansible_memtotal_mb >= minimal_master_memory_mb
-  when:
-    - not ignore_assert_errors
-    - ('kube_control_plane' in group_names)
-
-- name: Stop if memory is too small for nodes
-  assert:
-    that: ansible_memtotal_mb >= minimal_node_memory_mb
+    that: groups.get('etcd', groups.kube_control_plane) | length is not divisibleby 2
```

Comment:
group control plane can be even

Reply:
Not if there is no etcd group, no ? In that case we consider etcd == kube_control_plane so kube_control_plane would need to be odd.
(If I'm missing something I'll just use `['']` as default instead)

Reply:
thanks, all good